### PR TITLE
issue/2107/remove deprecatedBy

### LINKED ISSUE #2107: Remove deprecatedBy from properties
Yes, I want to get rid of this code as well.

_Originally posted by @externl in https://github.com/zeroc-ice/ice/pull/2100#discussion_r1588013871_

Our Ice property code currently supports deprecating one property by another, with the assumption they're still "compatible". We then also perform an extra check when setting a property which is deprecated by another one. If "Foo" is deprecated by "Bar", we actually set "Bar" (silently) but warning that "Foo" is deprecated.  

I think we should remove this functionality. 
1. Deprecating something means it still works for now but will be removed in the future. We actually just set a different property making it confusing if you attempt to look this property up in the future. 
2. If we do want to deprecate a property we should mark it as deprecated and still actually check it in code. In a future release we would remove it and its use entirely.

### PR DESCRIPTION
**Pull request by externl**
This PR removes support for the `deprecatedBy` feature of the Ice properties.

Fixes #2107

### CODE REVIEW

File: csharp/src/Ice/Properties.cs [hunk 1]
```diff
@@ -348,14 +348,8 @@ public void setProperty(string key, string val)
...
-        // Find the property, log warnings if necessary
-        Property? prop = findProperty(key, true);
-
-        // If the property is deprecated by another property, use the new property key
-        if (prop != null && prop.deprecatedBy != null)
-        {
-            key = prop.deprecatedBy;
-        }
+        // Checks if the property is a known Ice property and logs warnings if necessary
+        findProperty(key, true);
```
Comment (pepone) [inline csharp/src/Ice/Properties.cs:352]: calling a "findXxx" for side effects is not very nice.

**Review by pepone — APPROVED** [review]

File: csharp/src/Ice/Properties.cs (hunk 1 above)
Comment (externl) [reply csharp/src/Ice/Properties.cs:352]: I agree. We could rename it to something like `checkProperty`, but I'm note sure that's much better. 

We can only split one of the warnings out of the `findProperty` method. The other two depend on the knowledge that we have a known Ice prefix or not.

Any ideas?

File: config/makeprops.py [hunk 2]
```diff
@@ -27,7 +27,7 @@
     + "\n\n"
     + "// Generated by "
     + progname
-    + " from file %(inputfile)s, "
+    + " from file %(inputFile)s, "
```
Comment (bernardnormier) [inline config/makeprops.py:30]: I don't understand the trailing "s" here.

File: cpp/src/Ice/Properties.cpp [hunk 3]
```diff
@@ -106,19 +100,12 @@ namespace
...
-            for (int j = 0; j < IceInternal::PropertyNames::validProps[i].length; ++j)
-            {
-                const IceInternal::Property& prop = IceInternal::PropertyNames::validProps[i].properties[j];
-                if (IceUtilInternal::match(string{key}, prop.pattern))
-                {
-                    // There's always a non-null default value.
-                    return prop.defaultValue;
-                }
-            }
+            throw invalid_argument{"unknown ice property: " + string{key}};
```
Comment (bernardnormier) [inline cpp/src/Ice/Properties.cpp:106]: Ice should be capitalized.

**Review by bernardnormier — APPROVED** [review]

File: config/makeprops.py (hunk 2 above)
Comment (pepone) [reply config/makeprops.py:30]: This old style string format in Python

- `%` is the string formatting operator.
- `s` specifies that the value should be formatted as a string.

We should switch to f""" {inputFile}"""

File: java/src/Ice/src/main/java/com/zeroc/Ice/Properties.java [hunk 4]
```diff
@@ -347,12 +347,14 @@ public void setProperty(String key, String value) {
       throw new InitializationException("Attempt to set property with empty key");
     }
 
-    // Find the property, log warnings if necessary
+    // Finds the corresponding Ice property if it exists. Also logs warnings for unknown Ice
+    // properties and
+    // case-insensitive Ice property prefix matches.
```
Comment (pepone) [inline java/src/Ice/src/main/java/com/zeroc/Ice/Properties.java:352]:
```suggestion
    // properties and case-insensitive Ice property prefix matches.
```

**Review by pepone — APPROVED** [review]

File: config/makeprops.py (hunk 2 above)
Comment (externl) [reply config/makeprops.py:30]: I'll clean this up in a following PR

**Pull request merged**